models/podcastDbModel.js

```diff
@@ -0,0 +1,43 @@
+const mysql = require('../utils/mysql')
+
+const getPodcastsItems = async () => {
+  return await mysql.runQuery('SELECT * FROM `podcasts`.`podcasts`;')
+}
+
+const searchItem = async (queryParams) => {
+  const query = `%${queryParams}%`
+  const searchResult = await mysql.runQuery('SELECT * FROM `podcasts`.`podcasts` WHERE title OR author LIKE ?;', query)
+  if (!searchResult.length) {
+    return null
+  } else {
+    return searchResult
+  }
+}
+
+const getItem = async (id) => {
+  return await JSON.parse(JSON.stringify(await mysql.runQuery('SELECT * FROM `podcasts`.`podcasts` WHERE id=?;', [id])))[0]
+}
+
+const updateItem = async (id, podcast) => {
+  return await mysql.runQuery('UPDATE `podcasts`.`podcasts` SET title=?,author=?,description=?,htmlDescription=?,webUrl=?,imageUrl=?,language=?,numberOfEpisodes=?,avgEpisodeLength=?,category=? WHERE id=?;',
```

Comment:
what if you are getting only a part of the fields and not all of them?

Reply:
I address this case inside the service function (by fetching the old podcast and appending the changes to it before  updating it).

Reply:
fixed by changing the query using "COALESCE"